### ISSUE TITLE
Multi-worker dyn batching

### ISSUE THREAD

**Issue by Coach257** [ByteDance-Seed/VeOmni#511]
Merge #488  in this commit: [Merge remote-tracking branch 'origin/main' into verl_trainer](https://github.com/ByteDance-Seed/VeOmni/pull/458/commits/a72ce016023c06a2586b916030f9ef76695595c7)
Several works todo:
1. add args in arguments
2. add & test multiprocess dataloader building in task & trainer
3. use trainer in `test_dynamic_batching_dataset.py`, build the same data by using dummydataset and load data with datasets_type = mapping | iterable instead of using mappingdummydataset | iterabledummydataset.
@LiuzcEECS

_Originally posted by @Coach257 in https://github.com/ByteDance-Seed/VeOmni/issues/458#issuecomment-3958069529_

**Comment by LiuzcEECS** [comment]
1 is tracked in https://github.com/ByteDance-Seed/VeOmni/issues/491
all the updates can be done after this PR
About the mapping dataset, the build_mapping_dataset requires "creating dummydataset -> saving files -> loading files with build_mapping_dataset" which I feel redundant and unnecessary in testing, so I created DummyMappingDataset to make it easier, but I agree maybe that can be omitted.
About the iterable dataset, the build_iterable_dataset func with "iterable" datasets_type generates huggingface Iterable dataset, in which state_dict() cannot save buffer when shuffling, which caused a problem of testing saving/loading with dyn batching and multi-sourcing.
